specification/dartLangSpec.tex

```diff
@@ -20710,13 +20706,13 @@ \subsection{Static Types}
   \alt <typeNotFunction>
 
 <typeNotVoid> ::= <functionType> `?'?
-  \alt <typeNotVoidNotFunction>
+  \alt <typeNotVoidNotFunction> `?'?
 
 <typeNotFunction> ::= \VOID{}
-  \alt <typeNotVoidNotFunction>
+  \alt <typeNotVoidNotFunction> `?'?
 
-<typeNotVoidNotFunction> ::= <typeName> <typeArguments>? `?'?
-  \alt \FUNCTION{} `?'?
+<typeNotVoidNotFunction> ::= <typeName> <typeArguments>?
+  \alt \FUNCTION{}
```

Comment:
Is `Function` not already a `<typeName>`?

It's not a built-in identifier.

Reply:
`Function` is actually a built-in identifier (since https://dart-review.googlesource.com/c/sdk/+/41360).